src/redis-cli.c

```diff
@@ -2385,6 +2390,12 @@ static int confirmWithYes(char *msg, int ignore_force) {
 
 static int issueCommandRepeat(int argc, char **argv, long repeat) {
     while (1) {
+        if (!config.eval_ldb && /* In debugging mode, let's pass "help" to Redis. */
+            (!strcasecmp(argv[0],"help") || !strcasecmp(argv[0],"?"))) {
+            cliOutputHelp(--argc, ++argv);
+            return REDIS_OK;
+        }
```

Comment:
let's move it outside the loop, and maybe add a comment that we can process this "command" without a connection.